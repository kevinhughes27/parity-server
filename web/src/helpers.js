import _ from 'lodash'

export function calcSalaryLimits(players) {
  const numTeams = 10
  const salaryCapVariance = 0.02
  const salaries = _.map(players, (p) => p.salary)
  const salaryAvg = _.sum(salaries) / numTeams;
  
  return {
    salaryCap: salaryAvg * (1 + salaryCapVariance),
    salaryFloor: salaryAvg * (1 - salaryCapVariance),
  }
}

export const colors = [
  '#C8E6C9',
  '#BDE0BE',
  '#B2DAB3',
  '#A7D4A8',
  '#9CCF9E',
  '#91C993',
  '#85C388',
  '#7ABD7D',
  '#6FB772',
  '#64B168',
  '#59AC5D',
  '#4EA652',
  '#43A047'
]

export const warnColors = [
  '#FFF9C4',
  '#FFF6B8',
  '#FFF4AC',
  '#FEF1A0',
  '#FEEE94',
  '#FEEB88',
  '#FEE87D',
  '#FEE671',
  '#FEE365',
  '#FDE059',
  '#FDDE4D',
  '#FDDB41',
  '#FDD835'
]

export const dangerColors = [
  '#FFCDD2',
  '#FDC1C5',
  '#FBB4B8',
  '#F8A8AB',
  '#F69C9E',
  '#F48F91',
  '#F28384',
  '#F07776',
  '#EE6A69',
  '#EB5E5C',
  '#E9524F',
  '#E74542',
  '#E53935'
]
