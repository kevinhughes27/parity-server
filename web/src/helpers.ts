import { map, sum, uniq } from 'lodash'

interface IPlayer {
  name: string;
  team: string;
  salary: number;
}

export function calcSalaryLimits(players: IPlayer[]) {
  const numTeams = uniq(players.map(p => p.team)).length
  const salaryCapVariance = 0.02
  const salaries = map(players,
     (p) => p.salary)
  const salaryAvg = sum(salaries) / numTeams;

  return {
    salaryCap: salaryAvg * (1 + salaryCapVariance),
    salaryFloor: salaryAvg * (1 - salaryCapVariance),
  }
}

// hex gradient maker:
// https://www.strangeplanet.fr/work/gradient-generator/index.php

export const colors = [
  '#C8E6C9',
  '#BDE0BE',
  '#B2DAB3',
  '#A7D4A8',
  '#9CCF9E',
  '#91C993',
  '#85C388',
  '#7ABD7D',
  '#6FB772',
  '#64B168',
  '#59AC5D',
  '#4EA652',
  '#43A047'
]

export const underColors = [
  '#bbdefb',
  '#abd3f5',
  '#9cc8f0',
  '#8dbdea',
  '#7eb2e5',
  '#6fa7e0',
  '#609cda',
  '#5191d5',
  '#4286d0',
  '#337bca',
  '#2470c5',
  '#1565c0',
]

export const overColors = [
  '#FFCDD2',
  '#FDC1C5',
  '#FBB4B8',
  '#F8A8AB',
  '#F69C9E',
  '#F48F91',
  '#F28384',
  '#F07776',
  '#EE6A69',
  '#EB5E5C',
  '#E9524F',
  '#E74542',
  '#E53935'
]

export const homeColors = [
  "#90CAF9",
  "#84BEF1",
  "#78B2E9",
  "#6CA6E1",
  "#609AD9",
  "#548ED1",
  "#4882C9",
  "#3C76C1",
  "#306AB9",
  "#245EB1",
  "#1852A9",
  "#0D47A1"
]

export const awayColors = [
  "#FFCC80",
  "#FDC374",
  "#FCBA68",
  "#FAB15D",
  "#F9A951",
  "#F7A045",
  "#F6973A",
  "#F48E2E",
  "#F38622",
  "#F17D17",
  "#F0740B",
  "#EF6C00"
]
