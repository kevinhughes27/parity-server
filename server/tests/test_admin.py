from .helpers import get_stats, upload_game
from pathlib import Path
import json


def test_auth(client):
    response = client.post("/api/1/games/1")
    assert response.status_code == 401

    response = client.delete("/api/1/games/1")
    assert response.status_code == 401


def edit_game(client, league, game_id, data_file):
    fixture_path = Path(__file__).parent / "data" / data_file

    with open(fixture_path) as f:
        game_str = f.read()

    game = json.loads(game_str)

    response = client.post(
        f"/api/{league.id}/games/{game_id}", json=game, auth=("admin", "testpw")
    )
    assert response.status_code == 200, response.json()
    assert response.text == '"OK"'


def test_edit(client, league, rosters, monkeypatch, snapshot):
    upload_game(client, "mini_game.json")

    initial_stats = get_stats(client)
    assert initial_stats["stats"]["Brian Kells"]["pulls"] == 1
    assert initial_stats["stats"]["Scott Higgins"]["pulls"] == 0

    monkeypatch.setenv("PARITY_EDIT_PASSWORD", "testpw")

    game_id = 1
    edit_game(client, league, game_id, "mini_game_edited.json")

    stats = get_stats(client)
    assert stats["stats"]["Brian Kells"]["pulls"] == 0
    assert stats["stats"]["Scott Higgins"]["pulls"] == 1

    assert stats == snapshot


def test_delete(client, league, rosters, monkeypatch, snapshot):
    upload_game(client, "mini_game.json")

    initial_stats = get_stats(client)
    assert initial_stats["stats"]["Brian Kells"]["pulls"] == 1
    assert initial_stats["stats"]["Scott Higgins"]["pulls"] == 0

    monkeypatch.setenv("PARITY_EDIT_PASSWORD", "testpw")

    game_id = 1
    response = client.delete(
        f"/api/{league.id}/games/{game_id}", auth=("admin", "testpw")
    )
    assert response.status_code == 200, response.json()
    assert response.text == '"OK"'

    response = client.get("/api/1/stats")
    assert response.status_code == 200
    assert len(response.json()["stats"]) == 0
