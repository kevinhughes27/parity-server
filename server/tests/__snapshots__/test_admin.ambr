# serializer version: 1
# name: test_edit
  dict({
    'stats': dict({
      'Ashlin Kelly': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Benjamin Piper': dict({
        'assists': 1,
        'callahan': 0,
        'catches': 2,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 0,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.5,
        'o_points_against': 1,
        'o_points_for': 1,
        'pay': 11500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 5750,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 1,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Brent Burton': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Brian Kells': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 9500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 4750,
        'second_assists': 1,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Carrie-Anne Whyte': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 0,
        'd_points_for': 0,
        'drops': 1,
        'goals': 0,
        'o_efficiency': 0.5,
        'o_points_against': 1,
        'o_points_for': 1,
        'pay': -3000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': -1500,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Chris Sullivan': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.5,
        'd_points_against': 1,
        'd_points_for': 1,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.0,
        'o_points_against': 0,
        'o_points_for': 0,
        'pay': 10500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 5250,
        'second_assists': 1,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Christine Beals': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Christopher Keates': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 2,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.5,
        'd_points_against': 1,
        'd_points_for': 1,
        'drops': 0,
        'goals': 2,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 32500,
        'points_played': 3,
        'pulls': 0,
        'salary_per_point': 10833,
        'second_assists': 1,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.6666666666666666,
      }),
      'Jason Fraser': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Jessie Robinson': dict({
        'assists': 1,
        'callahan': 0,
        'catches': 1,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.5,
        'd_points_against': 1,
        'd_points_for': 1,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.0,
        'o_points_against': 0,
        'o_points_for': 0,
        'pay': 13000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 6500,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Jim Robinson': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 1,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 0,
        'd_points_for': 0,
        'drops': 0,
        'goals': 1,
        'o_efficiency': 0.5,
        'o_points_against': 1,
        'o_points_for': 1,
        'pay': 11500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 5750,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Jonathan Champagne': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Justine Price': dict({
        'assists': 1,
        'callahan': 0,
        'catches': 1,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 12000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 6000,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Krys Kudakiewicz': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.5,
        'd_points_against': 1,
        'd_points_for': 1,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.0,
        'o_points_against': 0,
        'o_points_for': 0,
        'pay': 2000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 1000,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Martin Cloake': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 1,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 1,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 11500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 5750,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Meagan Doyle': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Michelle Warren': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.5,
        'd_points_against': 1,
        'd_points_for': 1,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.0,
        'o_points_against': 0,
        'o_points_for': 0,
        'pay': 2000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 1000,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Patrick McKelvey': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 1,
        'completions': 2,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 0,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.5,
        'o_points_against': 1,
        'o_points_for': 1,
        'pay': 10500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 5250,
        'second_assists': 1,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Richard Gregory': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Rob Ives': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.5,
        'd_points_against': 1,
        'd_points_for': 1,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.0,
        'o_points_against': 0,
        'o_points_for': 0,
        'pay': 2000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 1000,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Sam Lee': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 0,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.5,
        'o_points_against': 1,
        'o_points_for': 1,
        'pay': 1000,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 500,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Scott Higgins': dict({
        'assists': 1,
        'callahan': 0,
        'catches': 1,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 1.0,
        'o_points_against': 0,
        'o_points_for': 1,
        'pay': 12000,
        'points_played': 2,
        'pulls': 1,
        'salary_per_point': 6000,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
      'Simon Berry': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 0,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 1,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.0,
        'o_points_against': 0,
        'o_points_for': 0,
        'pay': 0,
        'points_played': 1,
        'pulls': 0,
        'salary_per_point': 0,
        'second_assists': 0,
        'team': '99 Problems',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.0,
      }),
      'Tim Kealey': dict({
        'assists': 0,
        'callahan': 0,
        'catches': 0,
        'completions': 1,
        'd_blocks': 0,
        'd_efficiency': 0.0,
        'd_points_against': 0,
        'd_points_for': 0,
        'drops': 0,
        'goals': 0,
        'o_efficiency': 0.5,
        'o_points_against': 1,
        'o_points_for': 1,
        'pay': 1500,
        'points_played': 2,
        'pulls': 0,
        'salary_per_point': 750,
        'second_assists': 0,
        'team': 'Kells Angels Bicycle Club',
        'threw_drops': 0,
        'throw_aways': 0,
        'total_efficiency': 0.5,
      }),
    }),
    'week': 0,
  })
# ---
